File: ruby/WeaponType.rb
#encoding:utf-8

module Deepspace
    module WeaponType
        class Type
            def initialize (pot)
                @power = pot
            end

            def power
                @power
            end

            def to_s
                if (self == Deepspace::WeaponType::LASER)
                    t = "LASER"
                elsif (self == Deepspace::WeaponType::MISSILE)
                    t = "MISSILE"
                else
                    t = "PLASMA"
                end
                return t
            end
        end

        #las unicas instancias:
        LASER = Type.new(2.0)
        MISSILE = Type.new(3.0)
        PLASMA = Type.new(4.0)
        
    end
end

File: java/Deepspace/src/deepspace/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author alvaro
 */
public enum WeaponType {
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
    
    private float power;    //final???
    
    WeaponType(float p){
        power = p;
    }
    
    float getPower(){
        return power;
    }
    
    //LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f); //tiene que ir en la primera linea, si no da error
}

